specification/iotcentral/resource-manager/Microsoft.IoTCentral/stable/2018-09-01/iotcentral.json

```diff
@@ -338,20 +338,66 @@
             "$ref": "#/parameters/subscriptionId"
           },
           {
-            "name": "operationInputs",
+            "name": "nameAvailabilityInputs",
             "in": "body",
-            "description": "Set the name parameter in the OperationInputs structure to the name of the IoT Central application to check.",
+            "description": "Set the name parameter in the NameAvailabilityInputs structure to the name of the IoT Central application to check.",
             "required": true,
             "schema": {
-              "$ref": "#/definitions/OperationInputs"
+              "$ref": "#/definitions/NameAvailabilityInputs"
```

Comment:
Is this change of name necessary? This is a breaking change on the client side.

Reply:
Would it be better if I reverted the name back to `OperationInputs` and add the `subdomain` property to this model. The only problem I encountered when doing this was having set the `required` field to false for the `req.body`. Not sure if this is the best approach, thoughts?

Reply:
@r-delgadillo Adding a property that's not required is okay but changing a required property to optional is a breaking change.

Reply:
Why do you need to set the body parameter to be optional? Seems like the difference between `OperationInputs` and `NameAvailabilityInputs` is just an optional property `type`.

Reply:
@jianghaolu The reason for this change was to make it clear to the user that the two endpoints `checkNameAvailability` and `checkSubdomainAvailability` have different input parameters, one with `name` and the other `subdomain`. After discussing with the team, we decided to keep `OperationInputs` and pass this model into both endpoints `req.body`. Thanks for the feedback.